src/ui/units/dash/containers/Body/Body.tsx

```diff
@@ -1266,18 +1303,19 @@ class Body extends React.PureComponent<BodyProps, DashBodyState> {
                     <div
                         className={b('content', {
                             condensed: this.isCondensed(),
-                            'with-table-of-content': showTableOfContent && hasTableOfContent,
+                            'with-table-of-content': showTableOfContent && hasTableOfContentForTab,
                             mobile: DL.IS_MOBILE,
                             aside: getIsAsideHeaderEnabled(),
                             'with-edit-panel': showEditActionPanel,
                             'with-footer': isEnabledFeature(Feature.EnableFooter),
                         })}
                     >
-                        {!settings.hideDashTitle && !DL.IS_MOBILE && (
-                            <div className={b('entry-name')} data-qa={DashEntryQa.EntryName}>
-                                {Utils.getEntryNameFromKey(this.props.entry?.key)}
-                            </div>
-                        )}
+                        {!settings.hideDashTitle &&
+                            (!DL.IS_MOBILE || isMobileFixedHeaderEnabled) && (
```

Comment:
Let's move this bool check into variable before render like `hasTableOfContentForTab`